utils/manifest-utils/src/rules.rs

```diff
@@ -0,0 +1,146 @@
+use std::{
+    fs::OpenOptions,
+    io::BufReader,
+    path::{Path, PathBuf},
+    sync::{Arc, RwLock},
+};
+
+use anyhow::Result;
+use serde::{Deserialize, Serialize};
+use structopt::StructOpt;
+
+#[derive(Clone, Debug, Default)]
+pub struct RuleStore {
+    pub path: PathBuf,
+    config: Arc<RwLock<RulesConfig>>,
+}
+
+impl RuleStore {
+    pub fn load_or_create(rules_file: &Path) -> Result<Self> {
+        if rules_file.exists() {
+            log::debug!("Loading rule from: {:?}", rules_file);
+            let file = OpenOptions::new().read(true).open(rules_file)?;
+
+            Ok(Self {
+                config: Arc::new(serde_json::from_reader(BufReader::new(file))?),
```

Comment:
Here is potential problem, when file is empty, but exists, this will fail.

Reply:
But this isn't the only place when we have this problem

Reply:
That's right. We should answer this question:
What if user deliberately makes invalid config or deletes all contents of a file?
Should we emit an error or maybe replace this config with default one?
IMO we shouldn't replace it as user can have very sophisticated config, he can edit it manually and when editing he can make a mistake and suddenly all of his work is deleted